Tests/SwiftDocCTests/Infrastructure/SymbolGraph/SymbolGraphRelationshipsBuilderTests.swift

```diff
@@ -153,4 +153,18 @@ class SymbolGraphRelationshipsBuilderTests: XCTestCase {
         // Test default implementation was added
         XCTAssertTrue((symbolIndex["A"]!.semantic as! Symbol).isRequired)
     }
+    
+    func testOptionalRequirementRelationship() throws {
+        let bundle = try testBundle(named: "TestBundle")
+        var symbolIndex = [String: DocumentationNode]()
+        let engine = DiagnosticEngine()
+        
+        let edge = createSymbols(in: &symbolIndex, bundle: bundle, sourceType: .init(parsedIdentifier: .class, displayName: "Class"), targetType: .init(parsedIdentifier: .protocol, displayName: "Protocol"))
```

Comment:
Nit: I know this is largely copied from the test above, and doesn't matter in the correctness of the test, but it's slightly weird to define the relationship from a _class_ to protocol. I'd expect the source to be a method.